src/library/setlogfeature.cpp

```diff
@@ -71,6 +71,10 @@ void SetlogFeature::bindWidget(WLibrary* libraryWidget,
 
 void SetlogFeature::bindSidebarWidget(WLibrarySidebar* pSidebarWidget) {
     m_pSidebarWidget = pSidebarWidget;
```

Comment:
Add here:
```
if (m_pMenu != nullptr) {
    delete m_pMenu;
}
```
and likewise for the other features

Reply:
Actually I don't think you need to delete the menu. I think using QWidget::setParent would work:
```
if (m_pMenu == nullptr) {
   m_pMenu = new QMenu();
}
m_pMenu->setParent(pSidebarWidget);
```

Reply:
for which case would this be useful?
now the menu is created only when there's a sidebar widegt passed from skinloader.
if the skin doesn't have a sidebar widget there's no sidebar bound to library.cpp > no sidebar passed on to library features, and there won't be any sidebar right-click either

Reply:
In case this `bindSidebarWidget` is ever called multiple times in the lifetime of the Feature. I don't think it is, but if that ever does happen in the future, the menu would have the wrong parent if this isn't changed.

Reply:
hmkay. so the correct snippet for _all features_ would be
```
if (m_pMenu == nullptr) {
    m_pMenu = new QMenu(pSidebarWidget);
}
m_pMenu->setParent(pSidebarWidget);
```
because with this
```
if (m_pMenu == nullptr) {
    m_pMenu = new QMenu();
}
...
```
the menu wouldn't have a parent and there wouldn't be a way to style it, like before this PR

Reply:
```
if (m_pMenu == nullptr) {
   m_pMenu = new QMenu();
}
m_pMenu->setParent(pSidebarWidget);
```
sets the parent regardless of whether m_pMenu existed previously. I suppose it wouldn't hurt to specify the parent to the QMenu constructor, but I am not sure it's necessary either.

Reply:
forget my comment, I imagined an `else` at the start of the last line ;)